### PR TITLE
update oracleservice wallet

### PR DESCRIPTION
**Pull request by bettybao1209**

### CODE REVIEW

**Comment by cloud8little** [comment]
Set case2.json as the oracle node

config.json under neo-cli directory:

```
{
  "ApplicationConfiguration": {
    "Logger": {
      "Path": "Logs_{0}",
      "ConsoleOutput": true,
      "Active": true
    },
    "Storage": {
      "Engine": "LevelDBStore"
    },
    "P2P": {
      "Port": 51001,
      "WsPort": 51002
    },
    "UnlockWallet": {
      "Path": "wallet_1.json",
      "Password": "11111111",
      "StartConsensus": false,
      "IsActive": false
    },
    "PluginURL": "https://github.com/neo-project/neo-plugins/releases/download/v{1}/{0}.zip"
  }
}
```

config.json under Plugins\OracleService:

```
{
  "PluginConfiguration": {
    "Nodes": ["http://localhost:10332"],
    "Wallet": "case2.json",
    "MaxTaskTimeout": 432000000,
    "AllowPrivateHost": true,
    "AllowedContentTypes": [ "application/json" ],
    "Https": {
      "Timeout": 5000
    }
  }
}
```

start neo-cli, and input start oracle.

2882ea9b902257432ce31965031327f

**Comment by cloud8little** [comment]
Besides,  this need to be sync with the latest neo.

https://github.com/neo-project/neo-modules/blob/5a060a9a6c251a49e172a391f7e87ba695aa1316/src/OracleService/OracleService.cs#L343-L344

**Comment by bettybao1209** [comment]
> start neo-cli, and input start oracle.
> 
> 2882ea9b902257432ce31965031327f

The wallet should be the `current wallet` in cli rather than specified in oracle's config file.

**Review by shargon — DISMISSED** [review]

**Review by shargon — DISMISSED** [review]

**Comment by cloud8little** [comment]
retest.

**Comment by cloud8little** [comment]
Test Passed.
1. CN node, start oracle with opening the oracle node wallet, and then start oracle. deal with oracle request and persisting block.

f11e54855f20e764d9a509ffed3db25


2. Normal node. when config wallet is inactive, start oracle failed; then open wallet manually, start oracle successful.

491d84570ff0687c626b3e5a0657ac3

**Review by cloud8little — DISMISSED** [review]

**Review by chenzhitong — APPROVED** [review]

File: src/OracleService/OracleService.cs [hunk 1]
```diff
@@ -73,24 +74,16 @@ public override void Dispose()
...
-            if (password.Length == 0)
-            {
-                Console.WriteLine("Cancelled");
-                return;
-            }
 
-            wallet = new NEP6Wallet(Settings.Default.Wallet);
-            try
+            var walletProvider = GetService<IWalletProvider>();
+            if (walletProvider is null)
```
Comment (erikzhang) [inline src/OracleService/OracleService.cs:79]: It couldn't be `null`.

File: src/OracleService/OracleService.cs [hunk 2]
```diff
@@ -73,24 +74,16 @@ public override void Dispose()
...
-                wallet.Unlock(password);
-            }
-            catch (System.Security.Cryptography.CryptographicException)
-            {
-                Console.WriteLine($"Failed to open wallet");
+                Console.WriteLine("Please open wallet first!");
                 return;
             }
 
+            wallet = walletProvider.GetWallet();
```
Comment (erikzhang) [inline src/OracleService/OracleService.cs:85]: The `wallet` could be `null`.

**Review by erikzhang — CHANGES_REQUESTED** [review]

File: src/OracleService/OracleService.cs (hunk 1 above)
Comment (bettybao1209) [reply src/OracleService/OracleService.cs]: updated.

File: src/OracleService/OracleService.cs (hunk 2 above)
Comment (bettybao1209) [reply src/OracleService/OracleService.cs]: updated.

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**